### PR TITLE
BAEL-2738 - Convert String to JsonObject with GSON

### PR DESCRIPTION
**Pull request by dcalap**

### CODE REVIEW

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        Assert.assertTrue(jsonObject.get("name").getAsString().equals("Baeldung"));
+        Assert.assertTrue(jsonObject.get("java").getAsBoolean() == true);
+    }
+
+    @Test
+    void givenAStringAndGsonInstance_shouldConvertToJsonObject() throws Exception {
+        // Example 2: Using fromJson
+        String json = "{ \"name\": \"Baeldung\", \"java\": true }";
+        JsonObject convertedObject = new Gson().fromJson(json, JsonObject.class);
+        Assert.assertTrue(convertedObject.isJsonObject());
```
Comment (KevinGilmore) [inline gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]: let's include the other assertions here as well (as in lines 15-16 above)

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+public class JsonObjectConversionsUnitTest {
+
+    @Test
+    void givenAStringAndJsonParser_shouldConvertToJsonObject() throws Exception {
+        // Example 1: Using JsonParser
+        String json = "{ \"name\": \"Baeldung\", \"java\": true }";
+        JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
```
Comment (KevinGilmore) [inline gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]: Let's add a blank line before and after this one. This separates the given/when/then parts of the test.

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        Assert.assertTrue(jsonObject.isJsonObject());
+        Assert.assertTrue(jsonObject.get("name").getAsString().equals("Baeldung"));
+        Assert.assertTrue(jsonObject.get("java").getAsBoolean() == true);
+    }
+
+    @Test
+    void givenAStringAndGsonInstance_shouldConvertToJsonObject() throws Exception {
+        // Example 2: Using fromJson
+        String json = "{ \"name\": \"Baeldung\", \"java\": true }";
+        JsonObject convertedObject = new Gson().fromJson(json, JsonObject.class);
```
Comment (KevinGilmore) [inline gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]: Blank lines before and after here as well.

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package org.baeldung.gson.conversion;
+
+import com.google.gson.*;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+public class JsonObjectConversionsUnitTest {
+
+    @Test
+    void givenAStringAndJsonParser_shouldConvertToJsonObject() throws Exception {
```
Comment (KevinGilmore) [inline gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]:
```suggestion
    void whenUsingJsonParser_thenConvertToJsonObject() throws Exception {
```

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        // Example 1: Using JsonParser
+        String json = "{ \"name\": \"Baeldung\", \"java\": true }";
+        JsonObject jsonObject = new JsonParser().parse(json).getAsJsonObject();
+        Assert.assertTrue(jsonObject.isJsonObject());
+        Assert.assertTrue(jsonObject.get("name").getAsString().equals("Baeldung"));
+        Assert.assertTrue(jsonObject.get("java").getAsBoolean() == true);
+    }
+
+    @Test
+    void givenAStringAndGsonInstance_shouldConvertToJsonObject() throws Exception {
```
Comment (KevinGilmore) [inline gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]:
```suggestion
    void whenUsingGsonInstanceFromJson_thenConvertToJsonObject() throws Exception {
```

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java (hunk 2 above)
Comment (dcalap) [reply gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]: Done.

File: gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java (hunk 3 above)
Comment (dcalap) [reply gson/src/test/java/org/baeldung/gson/conversion/JsonObjectConversionsUnitTest.java]: Done.

**Pull request merged**